### DISCUSSION TITLE
how to intercept outgoing SQL statements using Quarkus + Hibernate Reactive (StatementInspector)

### DISCUSSION THREAD

**Discussion by jfsanchez91** [quarkusio/quarkus#44308] [Q&A]
How can I intercept (and modify) the outgoing SQL statements using Quarkus and Hibernate Reactive? I tried to do so by implementing a custom StatementInspector, but somehow the inspector is not executed, I guess (Quarkus) Hibernate Reactive does not follow the same principles here.

[a comment by an automated account is not shown]

**Comment by DavideD** [comment]
See https://stackoverflow.com/questions/76619519/sql-inspection-injection-in-hibernate-reactive

**Reply by DavideD** [reply]
I've updated the issue and added the workaround. But I'm not sure it will work for Quarkus. At the moment, I'm busy with the upgrade to ORM 7, but I will have a look at this when I've finished with it.

**Reply by jfsanchez91** [reply]
hi @DavideD , thanks for the insight. I already saw that workaround in SO :), but sadly it does not work with Quarkus, as Quarkus provides the `ExternalSqlClientPool` when setting up Hibernate (check `io.quarkus.hibernate.reactive.runtime.customized.QuarkusReactiveConnectionPoolInitiator`).
Somehow the `hibernate.vertx.pool.class` was never taken into consideration.

**Reply by DavideD** [reply]
I was afraid that was the case. Then I don't have a solution at the moment. Sorry.
